packages/components/src/internal/components/forms/detail/DetailPanelHeader.tsx

```diff
@@ -13,59 +13,50 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import React from 'react';
+import React, { FC, memo } from 'react';
 
 interface DetailPanelHeaderProps {
-    canUpdate: boolean;
     editing?: boolean;
     isEditable: boolean;
-    onClickFn?: () => void;
+    onClick?: () => void;
     title?: string;
-    useEditIcon: boolean;
     verb?: string;
     warning?: string;
 }
 
-export class DetailPanelHeader extends React.Component<DetailPanelHeaderProps, any> {
-    static defaultProps = {
-        title: 'Details',
-        useEditIcon: true,
-        verb: 'Editing',
-    };
-
-    render() {
-        const { isEditable, canUpdate, editing, onClickFn, warning, title, useEditIcon, verb } = this.props;
-
-        if (editing) {
-            return (
-                <>
-                    {verb} {title}
-                    <span className="detail__edit--heading">
-                        {warning !== undefined && (
-                            <span>
-                                <span> - </span>
-                                <span className="edit__warning">{warning}</span>
-                            </span>
-                        )}
-                    </span>
-                </>
-            );
-        }
+export const DetailPanelHeader: FC<DetailPanelHeaderProps> = memo(props => {
+    const { isEditable, editing, onClick, warning, title = 'Details', verb = 'Editing' } = props;
 
+    if (editing) {
         return (
-            <>
-                {title}
+            <div className="panel-heading">
+                {verb} {title}
                 <span className="detail__edit--heading">
-                    {isEditable && canUpdate && (
-                        <>
-                            <div className="detail__edit-button" onClick={onClickFn}>
-                                {useEditIcon ? <i className="fa fa-pencil-square-o" /> : 'Edit'}
```

Comment:
removing `useEditIcon` and just always using the icon instead of the text makes sense to me. Just curious if there were actually any usages that were passing `useEditIcon={false}`?

Reply:
Nope, that's why I made the change. We must have had something using that prop in the past, but I am not sure what.